### PR TITLE
Fix filter value type parsing and depreciate value_* fields

### PR DESCRIPTION
**Pull request by jharley**
After some testing with curl, it seemed if we properly parse-and-coerce the HCL-read string value that this would work fine without the type-specific `value_*` filter options while solving #102 in a round about fashion (and providing a much cleaner interface and nicer DX).

This also un-depreciates the filter's `value` while depreciating all of the `value_*` fields introduced by #29.

### CODE REVIEW

File: honeycombio/resource_board_test.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func testAccCheckBoardExists(t *testing.T, name string) resource.TestCheckFunc {
 			return fmt.Errorf("could not find created board: %w", err)
 		}
 
+		for i := range createdBoard.Queries {
```
Comment (jharley) [inline honeycombio/resource_board_test.go:103]: All changes in this file should have happened in #100.

I'm happy to move them to their own commit if you feel that's better.

**Review by dstrelau — APPROVED** [review]
